app/src/userpages/components/SnippetDialog/index.jsx

```diff
@@ -0,0 +1,153 @@
+// @flow
+
+import React, { Component, Fragment } from 'react'
+import { I18n, Translate } from 'react-redux-i18n'
+import copy from 'copy-to-clipboard'
+
+import Modal from '$shared/components/Modal'
+import Dialog from '$shared/components/Dialog'
+import Buttons from '$shared/components/Buttons'
+import DropdownActions from '$shared/components/DropdownActions'
+import CodeSnippet from '$shared/components/CodeSnippet'
+
+import { ProgrammingLanguages, StreamrClientRepositories } from '$shared/utils/constants'
+
+import styles from './snippetDialog.pcss'
+
+type Language = $Values<typeof ProgrammingLanguages>
+
+type GivenProps = {
+    name: string,
+    snippets: {
+        [Language]: string,
+    },
+    onClose: () => void,
+}
+
+type Props = GivenProps
+
+type State = {
+    selectedLanguage?: Language,
+    copied: boolean,
+}
+
+const SnippetLanguageMappings = {
+    [ProgrammingLanguages.JAVASCRIPT]: 'javascript',
+    [ProgrammingLanguages.JAVA]: 'java',
+}
+
+export class SnippetDialog extends Component<Props, State> {
+    state = {
+        selectedLanguage: undefined,
+        copied: false,
+    }
+
+    timeout: ?TimeoutID = null
+
+    componentWillUnmount() {
+        if (this.timeout) {
+            clearTimeout(this.timeout)
+        }
+    }
+
+    onSave = () => {}
+
+    onSelectLanguage = (language: Language) => {
+        this.setState({
+            selectedLanguage: language,
+            copied: false,
+        })
+    }
+
+    onCopy = () => {
+        const { snippets } = this.props
+        const snippetLanguages = Object.keys(snippets)
+        const selectedLanguage = this.state.selectedLanguage || (snippetLanguages.length > 0 && snippetLanguages[0])
+
+        if (selectedLanguage) {
+            copy(snippets[selectedLanguage])
+
+            this.setState({
+                copied: true,
+            })
+            if (this.timeout) {
+                clearTimeout(this.timeout)
+            }
+            this.timeout = setTimeout(() => {
+                this.setState({
```

Comment:
Maybe we should implement this `copy` functionality in a self-contained component? This pattern of controlling the `copy`/`copied` text appears to be occurring in at least a couple of different places.

Reply:
Yes, good idea! Were you thinking something like this?

```
  <CopyProvider>
  	{(copyFn, copied) => (
  		<button onClick={copyFn}>{copied ? 'Copy' : 'Copied'}</button>
  	)}
  </CopyProvider>
```

Reply:
That would work 👌The provider should also take care of handling the timeout.

Reply:
Might actually be more work than it initially seems :) I'll leave it out for now.